pkg/apis/druid/v1alpha1/druid_types.go

```diff
@@ -94,6 +94,8 @@ type DruidClusterSpec struct {
 	// Optional: toleration to be used in order to run Druid on nodes tainted
 	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
 
+	// Optional: affinity to be used to for enabling node, pod affinity and anti-affinity
+	Affinity *v1.Affinity `json:"affinity,omitempty"`
```

Comment:
can be added in the nodespec as well. This will be useful to support different Affinity specs across druid nodes.